R/standardize_MAE.R

```diff
@@ -277,3 +277,123 @@ refine_rowdata <- function(rd, se, default_v = "Undefined") {
   }
   rd
 }
+
+
+#' Set Unique Parental Identifiers
+#'
+#' This function sets the `CellLineName` field in 
+#' `colData` to be unique by appending the `clid` in parentheses for duplicates.
+#'
+#' @param se A SummarizedExperiment object.
+#' @return A SummarizedExperiment object with unique `CellLineName` in `colData`.
+#' @examples
+#' se <- SummarizedExperiment::SummarizedExperiment(
+#'   assays = list(counts = matrix(1:4, ncol = 2)),
+#'   colData = S4Vectors::DataFrame(CellLineName = c("ID1", "ID1"), clid = c("C1", "C2"))
+#' )
+#' se <- set_unique_cl_names(se)
+#' @export
+#' @keywords standardize_MAE
+set_unique_cl_names <- function(se) {
+  checkmate::assertClass(se, "SummarizedExperiment")
+  
+  col_data <- SummarizedExperiment::colData(se)
+  cl_name_idf <- get_env_identifiers("cellline_name")
+  clid_idf <- get_env_identifiers("cellline")
+  
+  if (!is.null(col_data[[cl_name_idf]])) {
+    duplicated_ids <- col_data[[cl_name_idf]][duplicated(col_data[[cl_name_idf]])]
+    if (length(duplicated_ids) > 0) {
+      for (dup_id in unique(duplicated_ids)) {
+        dup_indices <- which(col_data[[cl_name_idf]] == dup_id)
+        col_data[[cl_name_idf]][dup_indices] <- paste0(col_data[[cl_name_idf]][dup_indices],
+                                                       " (", col_data[[clid_idf]][dup_indices], ")")
+      }
+      SummarizedExperiment::colData(se) <- col_data
+    }
+  }
+  return(se)
+}
+
+#' Set Unique Drug Names
+#'
+#' This function sets the `DrugName`, `DrugName_2`, and `DrugName_3` fields in `rowData`
+#' to be unique by appending the corresponding `Gnumber`, `Gnumber_2`, and `Gnumber_3` in parentheses for duplicates.
+#'
+#' @param se A SummarizedExperiment object.
+#' @return A SummarizedExperiment object with unique `DrugName` fields in `rowData`.
+#' @examples
+#' se <- SummarizedExperiment::SummarizedExperiment(
+#'   assays = list(counts = matrix(1:9, ncol = 3)),
+#'   rowData = S4Vectors::DataFrame(DrugName = c("DrugA", "DrugA", "DrugB"),
+#'   Gnumber = c("G1", "G2", "G5"),
+#'   DrugName_2 = c("DrugC", "DrugC", "DrugD"),
+#'   Gnumber_2 = c("G3", "G4", "G5")
+#' ))
+#' se <- set_unique_drug_names(se)
+#' @export
```

Comment:
IMO This logic should be more sophisticated. Check:
```
 se <- SummarizedExperiment::SummarizedExperiment(
   assays = list(counts = matrix(1:9, ncol = 3)),
   rowData = S4Vectors::DataFrame(DrugName = c("DrugA", "DrugA", "DrugB"),
   Gnumber = c("G1", "G2", "G5"),
   DrugName_2 = c("DrugC", "DrugC", "DrugD"),
   Gnumber_2 = c("G3", "G3", "G5")
 ))
se <- set_unique_drug_names(se)
SummarizedExperiment::rowData(se)

DataFrame with 3 rows and 4 columns
     DrugName     Gnumber  DrugName_2   Gnumber_2
  <character> <character> <character> <character>
1  DrugA (G1)          G1  DrugC (G3)          G3
2  DrugA (G2)          G2  DrugC (G3)          G3
3       DrugB          G5       DrugD          G5
```
`DrugC` should stay as `DrugC` not as `DrugC (G3)`